dnf-behave-tests/features/module/switch.feature

```diff
@@ -0,0 +1,74 @@
+  | glob                            | node*                         |
+  | glob:glob                       | node*:*                       |
+  | glob:glob:glob                  | node*:*:*0801*                |
+
+@bz1809314
+Scenario: Switch a module stream that was only enabled
+   When I execute dnf with args "module enable nodejs:10"
+   Then the exit code is 0
+    And Transaction is following
+        | Action                   | Package            |
+        | module-stream-enable     | nodejs:10          |
+    And modules state is following
+        | Module    | State     | Stream    | Profiles  |
+        | nodejs    | enabled   | 10        |           |
+   When I execute dnf with args "module switch nodejs:8"
+   Then the exit code is 0
+    And Transaction is following
+        | Action                   | Package            |
+        | module-stream-switch     | nodejs:10 -> 8     |
+    And modules state is following
+        | Module    | State     | Stream    | Profiles  |
+        | nodejs    | enabled   | 8         |           |
+
+@bz1809314
+Scenario: I can switch a module stream, install packages from a new module and keep installed profile
+  Given I use repository "dnf-ci-fedora"
+   When I execute dnf with args "module install nodejs:8/minimal"
+   Then the exit code is 0
+    And Transaction contains
+        | Action                    | Package                                       |
+        | install-group             | nodejs-1:8.11.4-1.module_2030+42747d40.x86_64 |
+        | module-profile-install    | nodejs/minimal                                |
+    And modules state is following
+        | Module    | State     | Stream    | Profiles  |
+        | nodejs    | enabled   | 8         | minimal   |
+   When I execute dnf with args "module switch nodejs:10"
+   Then the exit code is 0
+    And Transaction contains
+        | Action                   | Package                                       |
+        | upgrade                  | nodejs-1:10.11.0-1.module_2200+adbac02b.x86_64 |
+        | upgrade                  | npm-1:10.11.0-1.module_2200+adbac02b.x86_64 |
+        | module-stream-switch     | nodejs:8 -> 10     |
+    And modules state is following
+        | Module    | State     | Stream    | Profiles       |
+        | nodejs    | enabled   | 10        |  minimal       |
+   When I execute dnf with args "module switch nodejs:8"
+   Then the exit code is 0
+    And Transaction contains
+        | Action                   | Package                                       |
+        | downgrade                | nodejs-1:8.11.4-1.module_2030+42747d40.x86_64 |
+        | downgrade                | npm-1:8.11.4-1.module_2030+42747d40.x86_64 |
+        | module-stream-switch     | nodejs:10 -> 8     |
+    And modules state is following
+        | Module    | State     | Stream    | Profiles       |
+        | nodejs    | enabled   | 8         |  minimal       |
```

Comment:
I know it may be hard to create some scenarios, but could you add at least some tests for error handling when the switch fails? At least non-existent module/stream should be trivial?

Reply:
ok

Reply:
I create an additional test. @lukash Please could you take a look?